DIPs/DIP1001.md

```diff
@@ -0,0 +1,58 @@
+# DoExpression
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 1001                                                            |
+| Author:         | [Cauterite](https://github.com/Cauterite)                       |
+| Implementation: | https://github.com/dlang/dmd/compare/master...Cauterite:doExpr0 |
+| Status:         |                                                                 |
+
+## Abstract
+
+A *DoExpression* `do(x,y,z)` is exactly equivalent to a *CommaExpression* `(x,y,z)`, but
+doesn't emit a deprecation warning.
+
+### Links
+
+[Updated grammar spec](https://github.com/dlang/dlang.org/compare/master...Cauterite:patch-1)
+
+[Deprecation of CommaExpression](https://dlang.org/deprecate.html#Using%20the%20result%20of%20a%20comma%20expression)
+
+## Description
+
+Using the result of a comma expression is scheduled for deprecation, so it would be
+nice to have new syntax which replaces it. *DoExpression* is a new *PrimaryExpression*
+which serves the purpose of *CommaExpression* with a more explicit syntax, avoiding
+the problems which lead to *CommaExpression*'s deprecation.
+
+### Rationale
+
+- Sometimes you still need to use a comma expression. We don't live in a perfect world where every function is pure,
+but we can live in a world where you don't need to split your expression into statements because of side-effects.
+- The `do` keyword is currently wasted on a single rarely-used statement type ('do-while' statements).
+Sure, you could implement this syntax with a variadic template function,
+but why waste this opportunity to recycle `do`, when it's the perfect keyword for this purpose?
+- Implementation is dead-simple (already finished).
+
+### Breaking changes / deprecation process
+
+None. Fully backwards-compatible.
+
+### Examples
+
+```d
+// same expression, different syntax
+assert((x, y, z) == do(x, y, z));
+
+auto x = do(); // Error: do-expression requires at least one argument
```

Comment:
This collides with function calls.  Can't you implement this in the standard library as a template?